File: news_rest/src/main/java/com/example/trade_centre/entity/Item.java
package com.example.trade_centre.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDate;

@Data
@Document(collection = "items")
public class Item {

    @Id
    private String id;
    @Field("itemName")
    private String itemName;
    @Field("basePrice")
    private Long basePrice;
    @Field("category")
    private Category category;
    @Field("amount")
    private int amount;
    @Field("level")
    private int level;
    @Field("owner")
    private User owner;
    @Field("sold")
    private boolean sold;
    @Field("buyer")
    private User buyer;
    @Field("dateCreated")
    private LocalDate dateCreated;
    @Field("slug")
    private String slug;
    @Field("trait")
    private Trait trait;
    @Field("quality")
    private Quality quality;
}

File: news-angular/src/app/models/item.ts
import { Category } from "./category";
import { Quality } from "./quality";
import { Trait } from "./trait";
import { User } from "./user";

export interface Item{
  id:any | null;
  itemName: String;
  basePrice: Number;
  category: Category;
  amount: Number;
  level: Number;
  owner: User;
  sold: boolean;
  dateCreated: Date;
  buyer: User | null;
  slug: String | null;
  trait: Trait;
  quality: Quality;
}
